common/scala/src/main/scala/org/apache/openwhisk/core/database/cosmosdb/CosmosDBArtifactStore.scala

```diff
@@ -223,11 +224,22 @@ class CosmosDBArtifactStore[DocumentAbstraction <: DocumentSerializer](protected
 
     val querySpec = viewMapper.prepareQuery(ddoc, viewName, startKey, endKey, realLimit, realIncludeDocs, descending)
 
+    val options = newFeedOptions()
+    val queryMetrics = if (transid.meta.extraLogging) {
+      options.setPopulateQueryMetrics(true)
+      Some(scala.collection.mutable.Buffer[QueryMetrics]())
```

Comment:
Any reason not to use a var for `queryMetrics`, init to `List.empty` (or whatever empty collection), and just assign to `queryMetrics = r.getQueryMetrics.values()` below, in case extra logging enabled?

Reply:
There can be multiple  `FeedResponse` with each having a set of `QueryMetrics` attached and they would need to be accumulated.  Hence need to use `Buffer` so as to append them and cannot use `List.empty`.

Reply:
Sorry, replace `List.empty` with `Buffer[QueryMetrics]()`.  I just thought the use of `Option` here was confusing - if you don't set `options.setPopulateQueryMetrics(true)`, does `r.getQueryMetrics.values()` have any values? 
If not, you can just always run `buffer.appendAll...`).
I guess not creating the buffer at all is slightly better, just harder to read.

Reply:
> I guess not creating the buffer at all is slightly better, just harder to read.

Yes the objective was to avoid that part for a debug logic. However I see the readability point. Refactored the code to use suggested approach